starter-code/src/App.js

```diff
@@ -1,18 +1,57 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React, { Component } from "react";
+// import logo from './logo.svg';
+import "./App.css";
+import contacts from "./contacts.json";
+var counter = 5;
+
+function selected(counter) {
+  const shuffled = contacts.sort(() => 0.5 - Math.random());
+  return shuffled.slice(0, counter);
+}
+var selectedArray = selected(counter);
```

Comment:
⚠️ Your array of contacts used for displaying should live in the state of your main components, instead of just living in the variable contacts. All the data that affects display should indeed be part of the state.